Services/PriceProvider.cs

```diff
@@ -30,13 +41,35 @@ public async Task AddPrices(IList<ShoppingCartItem> items)
             {
                 if (skuProducts.TryGetValue(item.ProductSku, out var product))
                 {
-                    var pricePart = product.ContentItem.As<PricePart>();
-                    if (pricePart != null)
+                    ContentItem contentItem = product.ContentItem;
+
+                    foreach (var pricePart in PartsOfType<PricePart>(contentItem))
                     {
-                        item.Prices.Add(new PrioritizedPrice(0, pricePart.Price));
+                        if (pricePart.Price.Currency == _moneyService.CurrentDisplayCurrency)
```

Comment:
This is making too strong an assumption on the business rules I think. Specifically, it assumes all products will have a price in the default currency (or something even stronger). I think if the currency selector selects no currency (null), then that means it has no preference, not that the default is the preference.

Actually it's worse than that. The semantics of price providers is that they are additive: their responsibility is to harvest all available prices for this product. Then it's the responsibility of price strategies to select which price should be used in the cart (and in checkout). So this logic should not be here at all.

Reply:
My thinking with the change to add the currency filtering here was to make sure only one currency is returned. The current price strategy has a requirement that only one currency is passed in which I think is a good pattern. If we return all currencies to the shopping cart it will have the responsibility to filter before passing to the price strategies.

The reason the default was used is that most sites will probably only support one currency. If more than one is supported there will most likely be the need to use a currency selector.

Reply:
Yeah, I think there's a bug then. I don't see any code doing that selection. Maybe there should be a currency parameter up-front for the whole thing. It's unclear to me how we should select what currency to use when there really are multiple currencies on the site. Need to think more about this.

Reply:
So if you look at the shopping cart controller, you'll see how it groups by currency, but before that, it is possible that prices with multiple currencies may exist on a product and the behavior then is unspecified and probably buggy. I created #56 to track the issue.